### PR TITLE
Added CurrentApiCompatibilityLevel property in TargetFrameworkResolver.cs, and updated obsolete API

### PR DESCRIPTION
**Pull request by ltmx**

### CODE REVIEW

File: src/NuGetForUnity/Editor/TargetFrameworkResolver.cs [hunk 1]
```diff
@@ -91,20 +96,12 @@ internal static class TargetFrameworkResolver
         };
 
         private static DotnetVersionCompatibilityLevel CurrentBuildTargetDotnetVersionCompatibilityLevel
-        {
-            get
+            => CurrentApiCompatibilityLevel switch
```
Comment (igor84) [inline src/NuGetForUnity/Editor/TargetFrameworkResolver.cs:99]: Aa far as I know switch expressions are not supported in Unity 2018 that we are still supporting. You can test this and other of your changes if you open src/NuGetForUnity.Tests project in Unity 2018.4.36f1.

**Comment by JoC0de** [comment]
Hi @LTMX 
thanks for the contribution.
Unfortunately we can't use the "newer" API until we increase the minimum required Unity version. see comment from @igor84 .
Regarding the code style changes, we are using ReSharper to enforce a consistent code formatting. You can run [pre-commit](https://pre-commit.com/) if you have it installed or run the formatting script manually by calling: `python tools/resharper-cleanupcode.py`. Alternatively I can also reformat the code.

**Comment by ltmx** [comment]
Thank you for this valuable information, I obviously missed, I will be mindful of this in the future !
Besides this, cheers !

**Comment by JoC0de** [comment]
@ltmx not your fault. It isn't easy to contribute to open source. Especially if it isn't easy to find what coding standards / style is used. You can't know what restrictions we have...
Most of your changes we already added to other PR's so should we just close this one?

**Comment by ltmx** [comment]
Concerning API changes on the side of unity, should I just do a conditional branch of the code to make the code future-proof in case Unity removes the old apis ?

**Comment by JoC0de** [comment]
@ltmx I added the "conditional branch" to #597
https://github.com/GlitchEnzo/NuGetForUnity/blob/06e4f77282e12534325d80ecf8d9702a30798a52/src/NuGetForUnity/Editor/TargetFrameworkResolver.cs#L116-L121

**Pull request closed without merging**